### ISSUE TITLE
Inference from DTU data

### ISSUE THREAD

**Issue by KevinCain** [kwea123/CasMVSNet_pl#7]
I see from 'dtu.py' expects the DTU training image format, e.g.:
`img_filename = os.path.join(self.root_dir, f'Rectified/{scan}/rect_{vid+1:03d}_{light_idx}_r5000.png')`

Also, I see for each image 'item' the code expects to load 'depth_visual' (PNG) and 'depth_map' (PFM).

Happily, I have no trouble invoking 'eval.py' to running inference on the DTU training images (640x512 PNG, e.g. 'rect_001_0_r5000.png'). I can obtain .ply files from 'eval.py' for all of the DTU scenes.

However, how can we substitute the training images for DTU full resolution images (1,600x1,200)?

I'm likely missing something easy. I've tried substituting a new scene in '.../datasets/lists/dtu/text.txt' and changing the above loading format to correspond to the provided full resolution DTU images (e.g., '00000001.PNG'):

`img_filename = os.path.join(self.root_dir, f'Rectified/{scan}/{vid+1:08d}.PNG')`

However, as we aren't provided with full resolution DTU 'depth_visual' (PNG) and 'depth_map' (PFM), loading halts, of course.

What do you suggest? I'm trying to reproduce your DTU results provided in the 'Release' links. Thanks!

**Comment by kwea123** [comment]
Hi, **inference** on any size is possible, you just need to specify `--img_wh 1600 1200` in `eval.py`.
For **training**, it is also possible, first download the full resolution depth from [here](https://virutalbuy-public.oss-cn-hangzhou.aliyuncs.com/share/cascade-stereo/CasMVSNet/dtu_data/dtu_train_hr/Depths_raw.zip) (also in the readme), and just manually add argument `img_wh=(1600,1200)` in each of the dataset:
https://github.com/kwea123/CasMVSNet_pl/blob/968b185518067d5b2c667adb9ea3c65477a77a21/train.py#L68-L77
Since every paper I read use 640x512 size for training, I fixed it from the beginning instead of making it a tunable parameter.

**Comment by KevinCain** [comment]
Thanks @kwea123 for your quick reply!

My root question was really just about your preferred way to point to source files. Let me rephrase. ;-)

I take it `eval.py` is what we should use for inference, but by default for DTU datasets, the code runs inference on the low-resolution training set and reads 'depth_visual' and 'depth_map' per image.

If we want instead to use the DTU testing images as input (which share the same cameras and `pair.txt`), or a novel group of images, what is the right way to do so?

**Comment by kwea123** [comment]
Do you want to eval on DTU or your customed dataset? My understanding is that you want to test on DTU, is that correct?
If you want to test DTU for arbitrary size, in `eval.py` the default test image size is set to 1152x864 which you can change, so it has nothing to do with low-resolution training set anymore.

**Comment by KevinCain** [comment]
To answer your question, I'd like to test with DTU first, then of course I have my own data (images, camera files, `pair.txt`) which I'd like to test with thereafter.

When I invoke `eval.py` with the default 1152x864 image size parameters and `test.txt` containing simply `scan1`, it loads the DTU training images, e.g.:
`.../mvs_training/dtu/Rectified/scan1_train/rect_001_0_r5000.png`

I can simply adapt `dtu.py` to point to the DTU testing set, but your code still expects 'depth_visual' and 'depth_map' per image, which makes me think I'm calling `eval.py` incorrectly.

Thanks again.

**Comment by kwea123** [comment]
If you use the default parameters in eval.py, it will load the **full resolution** image, not the **training** image (notice there is no `_train` in the folder name):
https://github.com/kwea123/CasMVSNet_pl/blob/968b185518067d5b2c667adb9ea3c65477a77a21/datasets/dtu.py#L161-L162

As for `depth_map` and `depth_visual`, I think they can be downloaded from the link I provide above, it contains all scans, including the test scans; but you're right, they're not necessary for inference. I will fix the code to not require `depth_visual` and `depth_map` in inference mode, for the moment can you download the files, or wait for me to correct the code?

**Comment by KevinCain** [comment]
Thank you @kwea123 for your last commits, which I can confirm work for me!

The only caveat for others following this thread is that the input names need to be conformed, as in the DTU dataset the training and test images have different naming conventions.
